src/analysis/souffle/examples/multimic/multimic-variant.arcs

```diff
@@ -0,0 +1,69 @@
+// To be able to handle this use-case, since we don't have information about 
+// policies or who is even in the room until runtime, we need to evaluate these 
+// checks at runtime. The taint analysis could work similarly to the static one 
+// we have now, but it happens at runtime: reading from handles causes the tags 
+// on that handle to propagate to the write handles. Downgrades still remove 
+// tags.
+
+// Represents an actual microphone within a smart mic listening to audio.
+particle MicIn
+  audio: writes AudioData { }
+  // MicIn says hasTag("audio", unidentified_people::userSpeech)
+  // NOTE: at runtime, this also receives facts that add tags for the 
+  // identified users from "MobileDeviceRuntime" (see also comment in 
+  // multimic.authlogic).
+
+// Storage credentials can be in the form of authorization logic facts
+// and guarded by the may/will check. So I didn't include credentials checking 
+// here.
+
+particle Notifier
+  audioIn: reads AudioData { }
+  networkChannelsToUserDevices: writes Notification { }
+  // (This could be more like "claims { ... } " to keep it more in Arcs style.)
+  // Notifier says {
+  //    will("deliver_privacy_notifications", "audioIn").
```

Comment:
I don't know that keeping Arcs style is a goal. We're using it as a starting point, but we're building something new.

One thing I do notice here is that we have a `Notifier says` block while enclosed within a `Notifier` particle. I wonder, in the future, if it would be more helpful to have an implicit speaker of the particle while enclosed in a particle's context. That said, super early for that kind of decision, just something to think about.

Reply:
I think making the speaker implicit is probably better like you said.

I also think syntax changes for aesthetics/convenience/readability are eventually going to be important, but like you said it is early to nail down those things down while we are still figuring out how to get it to work functionally. I left it totally verbose just so that what it means and how it connects with everything else mechanically is clear.